scripts/jenkins/ardana/ansible/roles/ardana_qe_tests/vars/keystone-api.yml

```diff
@@ -0,0 +1,18 @@
+---
+# Setting for Keystone-soapui
+SoapuiSettings: soapui-settings.xml
+
+keystone_soapui_code_folder: "{{ ardana_qe_tests_dir }}/ardana-qa-tests/keystone/testcase/functional/SoapUI"
+
+ardana_qe_test_get_failed_cmd: "cat {{ ardana_qe_test_log }} | grep FAIL | grep -v INFO | grep .txt | awk -F/ '{print $12}' | awk -F.txt '{print $1}'"
+
+ardana_qe_test_venv_requires:
+  - 'python-subunit'
+
+ardana_qe_tests_requires:
+  - "gcc"
+  - "python-devel"
+  - "java-1_8_0-openjdk-headless"
```

Comment:
You need to be careful when adding `java-1_8_0-openjdk-headless` here, there are tasks to remove those packages after running the test to keep the environment clean. However if you have an environment with integrated deployer `java-1_8_0-openjdk-headless` should already be installed and should not be removed.

I suggest that you keep track when installing those packages and only remove the ones that were installed (changed).

Reply:
You can record the task results from https://github.com/SUSE-Cloud/automation/blob/master/scripts/jenkins/ardana/ansible/roles/ardana_qe_tests/tasks/qe_test_requires.yml#L29 and add a condition to remove only the ones with the `changed` state.

Reply:
Updated